File: frontend/src/types/project.ts
export type ProjectStatus =
  | "created"
  | "analyzing"
  | "analyzed"
  | "guiding"
  | "completed"
  | "archived"

export interface Project {
  id: string
  name: string
  description: string | null
  status: ProjectStatus
  opportunity_id: string | null
  user_id: string
  progress: number
  current_step: string | null
  institution: string | null
  is_saved: boolean
  combined_ai_overview: string | null
  combined_ai_reading_tips: string | null
  created_at: string
  updated_at: string
}

export interface ProjectCreate {
  name: string
  description?: string
  opportunity_id?: string
}

export interface ProjectUpdate {
  name?: string
  description?: string
  status?: ProjectStatus
  is_saved?: boolean
}

export interface ProjectDocument {
  id: string
  project_id: string
  filename: string
  original_filename: string
  file_size: number | null
  file_hash: string | null
  status: string
  processing_progress: number
  page_count: number | null
  processed_pages: number
  chunk_count: number
  vectorized_chunk_count: number
  error_message: string | null
  is_scanned: boolean
  ocr_confidence: number | null
  original_language: string
  ai_overview: string | null
  ai_reading_tips: string | null
  detected_institution: string | null
  analysis_generated_at: string | null
  created_at: string
  updated_at: string
}

export interface DocumentSection {
  id: string
  bid_document_id: string
  section_type: string
  section_title: string | null
  section_number: string | null
  start_page: number
  end_page: number
  content_preview: string | null
  detected_by: string
  confidence: number | null
  ai_summary: string | null
  reading_guide: string | null
  created_at: string
}

File: backend/app/schemas/project.py
"""Project request/response schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class ProjectCreate(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    description: str | None = None
    opportunity_id: UUID | None = None
    institution: str = Field("adb", pattern="^(adb|wb|other)$")


class ProjectUpdate(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=200)
    description: str | None = None
    status: str | None = None
    progress: int | None = Field(None, ge=0, le=100)
    current_step: str | None = None
    is_saved: bool | None = None


class ProjectResponse(BaseModel):
    id: UUID
    name: str
    description: str | None = None
    status: str
    opportunity_id: UUID | None = None
    user_id: UUID
    progress: int
    current_step: str
    institution: str
    is_saved: bool
    combined_ai_overview: str | None = None
    combined_ai_reading_tips: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PaginatedProjects(BaseModel):
    items: list[ProjectResponse]
    total: int
    page: int
    page_size: int
    pages: int
